ManageRDSessions/ManageRDSessions.psm1

```diff
@@ -1,71 +1,108 @@
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_ACTIVE' -or $_.SessionState -eq 'STATE_IDLE' }
-    $sessions | Select-Object HostServer, UserName, UnifiedSessionId, SessionState | Format-Table
-}
+function Get-sbRDSession {
+    [OutputType('Custom.SB.RDSession')]
+    [cmdletbinding()]
+    param(
+        [Parameter(Mandatory = $false)]
+        [ValidateSet("Active", "Idle", "Connected", "Disconnected", "Any")]
+        [Alias("State")]
+        [string]$SessionState = "Any",
 
-function Get-RDDisconnectedSession {
-    [CmdletBinding()]
-    param (
-        [
-        Parameter(
-            Mandatory = $False,
-            HelpMessage = "Specify minimum number of idle session minutes"
-        )
-        ]
-        [Alias('idle')]
-        [int]$MinIdleTime = 0
+        [Parameter(Mandatory = $false)]
+        [int]$MinimumIdleTime = 0
     )
 
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_DISCONNECTED' -and ( $_.IdleTime / 60000 ) -gt $MinIdleTime }
-    $sessions | Select-Object HostServer, UserName, @{ Label = 'Idle Time (Mins)'; expression = {$_.IdleTime / 60000 -as [int] } }, UnifiedSessionId, SessionState | Format-Table
-}
+    Begin {
+        # Intentionally empty
+    }
+    # Process block is required when outputting objects to the pipeline, otherwise only the final object will be passed
+    Process {
 
-function Remove-RDDisconnectedSession {
-    [CmdletBinding()]
-    param (
-        [
-        Parameter(
-            Mandatory = $False,
-            HelpMessage = "Specify minimum number of idle session minutes"
-        )
-        ]
-        [Alias('idle')]
-        [int]$MinIdleTime = 0,
-        [switch]$Force = $False
-    )
+        $state = switch ($SessionState) {
+            "Active" { "STATE_ACTIVE" }
+            "Idle" { "STATE_IDLE" }
+            "Disconnected" { "STATE_DISCONNECTED" }
+            "Connected" { "STATE_CONNECTED" }
+            "Any" { "*" }
+        }
+        Write-Verbose "Querying RD Session Collection for $SessionState sessions"
+        $sessions = Get-RDUserSession | Where-Object {
+            $_.SessionState -like $state -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleTime
+        }
 
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_DISCONNECTED' -and ( $_.IdleTime / 60000 ) -gt $MinIdleTime }
+    foreach ($session in $sessions) {
+        # Creating PSObject to output and naming as my own type for use in later pipeline input...
+        $object = [PSCustomObject]@{
+            PSTypeName       = "Custom.SB.RDSession"
+            HostServer       = $session.HostServer
+            UserName         = $session.UserName
+            UnifiedSessionID = $session.UnifiedSessionID
+            SessionState     = $session.SessionState
+            IdleTime         = ($session.IdleTime / 60000 -as [int])
+        } #pscustomobject
 
-    foreach ( $session in $sessions ) {
+        # Outputting object
```

Comment:
I'd prob save some typing and do this here:
```
# Outputting object
[PSCustomObject] @{
    PSTypeName       = "Custom.SB.RDSession"
    HostServer       = $session.HostServer
    UserName         = $session.UserName
    UnifiedSessionID = $session.UnifiedSessionID
    SessionState     = $session.SessionState
    IdleTime         = ($session.IdleTime / 60000 -as [int])
}
```

Unless you're going to do something else with `$object`. If you were doing more with $object, I'd name it less generic; maybe `$CustomSBRDSession`